widget/table.go

```diff
@@ -121,6 +121,7 @@ func NewTableWithHeaders(length func() (int, int), create func() fyne.CanvasObje
 	t := NewTable(length, create, update)
 	t.ShowHeaderRow = true
 	t.ShowHeaderColumn = true
+	t.ExtendBaseWidget(t)
```

Comment:
This change breaks the ability to extend table doesn't it?
We use the extend-later pattern in all other widgets.

Reply:
In `NewTable`, we are calling it first. It should be in one place only or it lead to a wrong order in lock usage. If the right place is in `CreateRenderer` then I can remove it from `NewTable` and that also solve the problem. As NewTableWithHeaders was new, I assumed that it was the one to break past behavior here.

Reply:
I think the pattern we use is that it is CreateRenderer and also in MinSize (because sometimes you need it extended before rendering).